sgpt/app.py

```diff
@@ -71,7 +71,13 @@ def main(
         return
 
     if not prompt and not editor:
-        raise MissingParameter(param_hint="PROMPT", param_type="string")
+        if not sys.stdin.isatty():
+            prompt = sys.stdin.read()
+        else:
+            raise MissingParameter(param_hint="PROMPT", param_type="string")
+    elif prompt and not sys.stdin.isatty():
+        stdin_data = sys.stdin.read()
+        prompt = f"{stdin_data.strip()}\n{prompt}"
```

Comment:
```py
elif prompt and not sys.stdin.isatty():
```

Do we really need this second case?

If the user has provided a prompt, then why would we overwrite it with the standard input?

I think skipping this `elif` block also solves potential issues when stdin is not a TTY. (I removed it in my second patch.)

Reply:
Do you have some example uses for this feature?

I can imagine it might be useful like this:

```shell
$ sgpt 'Do you see any errors in this logfile?' - < /var/log/Xorg.0.log
```

But in that case, we'd probably want to put the `prompt` before the `stdin_data`

Reply:
Yes, using the positioning of `-` to place the stdin relative to the prompt could be quite intuitive. Unfortunately my implementation didn't do that!

Your thought made me imagine a possible approach in the future, for multiple files: https://github.com/TheR1D/shell_gpt/discussions/98

Anyway, for now I will withdraw my suggestion to use `-` because I see both you and Shakahs find the automatic detection of stdin more convenient than specifying `-`